src/plugins/add-cal/add-calendar.js

```diff
@@ -0,0 +1,155 @@
+	*/
+	init = function( event ) {
+
+		// Start initialization
+		// returns DOM object = proceed with init
+		// returns undefined = do not proceed with init (e.g., already initialized)
+		var elm = wb.init( event, componentName, selector ),
+			$elm = $( elm );
+
+		if ( elm ) {
+
+			wb.ready( $( elm ), componentName );
+
+			var properties = elm.querySelectorAll( "[property]" ),
+				event_details = new Object(),
+				i,
+				i_len,
+				prop_cache,
+				googleLink,
+				icsFile,
+				i18nDict = {
+					en: {
+						"addcal-addto": "Add to",
+						"addcal-calendar": "calendar",
+						"addcal-ical": "iCal format (iPhone, Outlook...)"
+					},
+					fr: {
+						"addcal-addto": "Ajouter au",
+						"addcal-calendar": "calendrier",
+						"addcal-ical": "Format iCal (iPhone, Outlook....)"
+					}
+				};
+
+			// Initiate dictionary
+			i18nDict = i18nDict[ $( "html" ).attr( "lang" ) || "en" ];
+			i18nDict = {
+				addto: i18nDict[ "addcal-addto" ],
+				calendar: i18nDict[ "addcal-calendar" ],
+				ical: i18nDict[ "addcal-ical" ]
+			};
+
+			//Set date stamp with the date modified
+			if ( $( "time[property='dateModified'][datetime]" ).length ) {
+				event_details.dtStamp = new Date( $( "time[property='dateModified']" ).attr( "datetime" ) );
+			} else {
+				event_details.dtStamp = new Date( $( "time[property~='dateModified']" ).text() );
+			}
+
+			event_details.dtStamp = event_details.dtStamp.toISOString().replace( /\..*[0-9]/g, "" ).replace( /-|:|\./g, "" );
+
+			i_len = properties.length;
+			for ( i = 0; i < i_len; i++ ) {
+				prop_cache = properties[ i ];
+				switch ( prop_cache.getAttribute( "property" ) ) {
+				case "name":
+					if ( $( prop_cache ).parentsUntil( ( "." + componentName ), "[typeof=Place]" ).length ) {
+						event_details.place = prop_cache.textContent;
+					} else {
+						event_details.name = prop_cache.textContent;
+					}
+					break;
+				case "description":
+					event_details.description = prop_cache.textContent.replace( /(\r\n|\n|\r)/gm, " " );
+					break;
+				case "startDate":
+					event_details.sDate = new Date( prop_cache.getAttribute( "datetime" ) ).toISOString().replace( /\..*[0-9]/g, "" ).replace( /-|:|\./g, "" );
+					break;
+				case "endDate":
+					event_details.eDate = new Date( prop_cache.getAttribute( "datetime" ) ).toISOString().replace( /\..*[0-9]/g, "" ).replace( /-|:|\./g, "" );
+					break;
+				case "location":
+					if ( !prop_cache.getAttribute( "typeof" ) ) {
+						event_details.place = prop_cache.textContent;
+					}
+					break;
+				case "streetAddress":
+					event_details.place += " " + prop_cache.textContent;
+					break;
+				case "addressLocality":
+					event_details.place += " " + prop_cache.textContent;
+					break;
+				case "addressRegion":
+					event_details.place += " " + prop_cache.textContent;
+					break;
+				case "postalCode":
+					event_details.place += " " + prop_cache.textContent;
+					break;
+				}
```

Comment:
Instead of compiling that string here with no order, I suggest to grab the info and then build the `place` string before to pre-build the ics file. At the same time, you will be able to apply the logic that would alternate when the "location" is here but with no typeof=Place